### DISCUSSION TITLE
questions regarding performance

### DISCUSSION THREAD

**Discussion by ddengster** [umputun/remark42#1459] [Q&A]
Great project, however, https://remark42.com/demo/ takes several seconds to load. There's quite a number of comments, some with images, so the load times could be understandable.

More questions regarding performance:
- I think it's written in a document somewhere that comments are written in an embedded file.  Is this done via Go and where's the source code that does that? 
- Are there alternatives if you want to switch backends for the comments?
- Can you disable images in comments?
- Is it possible to paginate comments? I can't find a `page` option in the api. Sorting by dates is not the most effective form of paging.

FWIW, I'm evaluating remark42 for a simple blog of mine, and generally want something lean, while this looks good it does feel slightly bloated due to the number of integrations.

**Comment by umputun** [comment]
> I think it's written in a document somewhere that comments are written in an embedded file. Is this done via Go and where's the source code that does that?

Data stored in embedded KV storage, see https://github.com/umputun/remark42/tree/master/backend/app/store/engine

> Are there alternatives if you want to switch backends for the comments?

You can implement a custom storage, see https://github.com/umputun/remark42/tree/master/backend/_example/memory_store

> Can you disable images in comments?

No

> Is it possible to paginate comments? I can't find a page option in the api. Sorting by dates is not the most effective form of paging.

sort of, but not really. For example `GET /api/v1/last/{max}?site=site-id&since=ts-msec` - get up to {max} last comments, since (epoch time, milliseconds) is optional allows some kind of pagination for latest comments, and   `/api/v1/list?site=site-id&limit=5&skip=2`- list commented posts, returns array or PostInfo, limit=0 will return all posts.

However, "find all comments for a given post" doesn't support any pagination

**Reply by ddengster** [reply]
Thanks for the quick reply! A few last questions:

>    Can you disable images in comments?
> No

Is this part of the markdown conversion that is hard to remove?

Is it possible to disable markdown for comments?

Also, would you be willing to accept PRs for these capabilities?

**Reply by paskal** [reply]
@ddengster, I understand the case for pagination, and we'll add it sooner or later, but what's the case for restricting users from adding images? Would a simple mode without formatting buttons be enough for you so that users would be able to embed images, but there would be no hints in the interface for them to do so?

**Reply by ddengster** [reply]
@paskal I don't want images possibly breaking my layouts, and my users to bother with that, and to give the impression of a lean blog site that requires little access control. Also there's the problem of uploading indecent images.

In any case, I've decided to implement my own frontend for learning purposes, with modified version of remark's backend. For images, it's a matter of throwing `bf.SkipImages` into the `FormatText` function. Will also trim all the stuff I don't need
